paasta_tools/vitesscluster_tools.py

```diff
@@ -15,6 +17,594 @@
+        "primary": 22505,
+        "read": 22291,
+    },
+    "refresh-service": {
+        "primary": 22790,
+        "read": 22687,
+    },
+    "refresh-nowaithost": {
+        "primary": 22351,
+        "read": 22160,
+    },
+    "refresh-nowaitcad": {
+        "primary": 22419,
+        "read": 22468,
+    },
+    "refresh-schematizer": {
+        "primary": 22840,
+        "read": 22827,
+    },
+    "sanitized-vitess": {
+        "primary": 23824,
+        "read": 23454,
+        "reporting": 23161,
+    },
+}
+
+# Environment variables
+VTCTLD_EXTRA_ENV = [
+    {
+        "name": "WEB_PORT",
+        "value": WEB_PORT,
+    },
+    {
+        "name": "GRPC_PORT",
+        "value": GRPC_PORT,
+    },
+    {
+        "name": "TOPOLOGY_FLAGS",
+        "value": "",
+    },
+]
+
+VTTABLET_EXTRA_ENV = [
+    {
+        "name": "CELL_TOPOLOGY_SERVERS",
+        "value": "",
+    },
+    {
+        "name": "SHARD",
+        "value": "0",
+    },
+    {
+        "name": "DB",
+        "value": "",
+    },
+    {
+        "name": "EXTERNAL_DB",
+        "value": "1",
+    },
+    {
+        "name": "KEYSPACE",
+        "value": "",
+    },
+    {
+        "name": "ROLE",
+        "value": "rdonly",
+    },
+    {
+        "name": "WEB_PORT",
+        "value": WEB_PORT,
+    },
+    {
+        "name": "GRPC_PORT",
+        "value": GRPC_PORT,
+    },
+    {
+        "name": "TOPOLOGY_FLAGS",
+        "value": "",
+    },
+    {
+        "name": "VAULT_ADDR",
+        "value": "https://vault-dre.uswest1-devc.yelpcorp.com:8200",
+    },
+    {
+        "name": "VAULT_ROLEID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vttablet-approle-roleid",
+                "key": "vault-vttablet-approle-roleid",
+            }
+        },
+    },
+    {
+        "name": "VAULT_SECRETID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vttablet-approle-secretid",
+                "key": "vault-vttablet-approle-secretid",
+            }
+        },
+    },
+    {
+        "name": "VAULT_CACERT",
+        "value": "/etc/vault/all_cas/acm-privateca-uswest1-devc.crt",
+    },
+]
+
+# Vault auth related variables
+VTGATE_EXTRA_ENV = [
+    {
+        "name": "VAULT_ADDR",
+        "value": "https://vault-dre.uswest1-devc.yelpcorp.com:8200",
+    },
+    {
+        "name": "VAULT_ROLEID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vtgate-approle-roleid",
+                "key": "vault-vtgate-approle-roleid",
+            }
+        },
+    },
+    {
+        "name": "VAULT_SECRETID",
+        "valueFrom": {
+            "secretKeyRef": {
+                "name": "paasta-vitessclusters-secret-vitess-k8s-vault-vtgate-approle-secretid",
+                "key": "vault-vtgate-approle-secretid",
+            }
+        },
```

Comment:
Please correct me if I'm wrong, but we're not planning to use vault secrets as mount path on the pods (vtgate, vttablet), but instead are directly reading from vault for each auth operation. The vault path (currently named `secrets/vitess/vt-tablet/vttablet_credentials.json` or `secrets/vitess/vt-gate/vttablet_credentials.json`) with secrets will get new or updated secrets/credentials on frequent basis outside of the deployment. But, the role id and secret id wouldn't change, and are needed to be setup only during the initial deployment unless we update the secret id itself?

Reply:
correct - but we generally want to avoid manually managed secrets: we don't currently do a full sync of secrets (as in, we don't delete unused data) , but in the future i'd like us to fully manage the secrets that we sync and ensure that k8s only has secrets that should actually exist

we've also discussed potentially sharding our EKS clusters, and having data like this that needs to be manually setup during initial deployments would complicate that since everyone involved would need to remember that there's a set of secrets that need to be manually created 

i think that as long as these secrets exist in the secrets/ directory, we'll be fine - barring a small window of time where it's possible for the secret sync to not have run on a fresh cluster/deployment, but this code has (which should be fine if i'm not mistaken: k8s should realize that the secrets eventually exist and everything should proceed)

Reply:
@flopex ^ Any thoughts on this?

Reply:
@flopex to confirm, but upon discussion it seems like the secret id and role id would not be modified when we rotate the secrets in vault periodically. So, there is no relevant paasta secret sync to be done as the vttablet and vtgate pods directly communicate with vault based on role id and secret id set before the first deployment in an ecosystem